client/client.go

```diff
@@ -509,6 +509,58 @@ func (r *NotaryRepository) GetChangelist() (changelist.Changelist, error) {
 	return cl, nil
 }
 
+// GetDelegationRoles returns the keys and roles of the repository's delegations
+func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
+	// Update state of the repo to latest
+	_, err := r.Update()
+	if err != nil {
+		return nil, err
+	}
+
+	// All top level delegations (ex: targets/level1) are stored exclusively in targets.json
+	targetsJSON, err := r.fileStore.GetMeta("targets", 0)
```

Comment:
After calling `r.Update()`, the validated repository is already de-serialized into `r.tufRepo`. No need to load the files from the filestore.